src/passes/EmscriptenPIC.cpp

```diff
@@ -110,34 +93,8 @@ struct EmscriptenPIC : public WalkerPass<PostWalker<EmscriptenPIC>> {
     Block* block = builder.makeBlock();
     assignFunc->body = block;
 
-    bool hasSingleMemorySegment =
-      module->memory.exists && module->memory.segments.size() == 1;
-
     for (Global* g : gotMemEntries) {
-      // If this global is defined in this module, we export its address
-      // relative to the relocatable memory. If we are in a main module, we can
-      // just use that location (since if other modules have this symbol too, we
-      // will "win" as we are loaded first). Otherwise, import a g$ getter. Note
-      // that this depends on memory having a single segment, so we know the
-      // offset, and that the export is a global.
       auto base = g->base;
```

Comment:
Sounds good to let wasm-ld do this work instead of us. Does it do the optimizations mentioned in these comments?

Reply:
Essentially yes, it avoid the GOT.mem or GOT.func import when we are the main module and that symbol is defined in the main module.

This means that gotMemEntries and gotFuncEntries for things we export will simply not be present anymore.

I will try replacing with an assert just to be 100% sure...